drafts/25-114.txt

```diff
@@ -246,6 +247,9 @@ FPP will not expand tokens in either fixed- or free-form:
     - In character literal constants.
     - In character string edit descriptors (quoted character strings).
 
+FPP will not expand tokens in any alternative syntax for character
+literal constants or edit descriptors that the processor supports.
```

Comment:
For the record, Malcolm's literal wording suggestion here was: (I took careful notes)
> “if a processor has an alternative syntax for character literal constants, macro expansion shall not take place within that literal constant”

I'm not opposed to massaging that wording, but I worry about the potential for unintended misreadings. For example, the text above could be misparsed/misread as "FPP will not expand tokens in ... edit descriptors that the processor supports" which is too broad and definitely not what we mean.

Do we really need to mention edit descriptors explicitly? Doesn't a Hollerith constant appearing in an edit descriptor still qualify as an "character literal constant (with alternate syntax)"?

```suggestion
literal constants that the processor supports.
```

Reply:
I was trying to mirror the text that preceded it for character literal constants and edit descriptors. The standard treats these as two different things, and calls them out separately. Now, the meaning is as if I could parenthesize this as "(character literal) constants or edit descriptors". The proper form is probably to hyphenate the compound adjective character-literal constants or edit descriptors.

Reply:
The potential misparse is to drop the entire phrase "alternative syntax for character literal constants" before the "or" and misread the second half as a prohibition on expansion in "any .. edit descriptors". I don't see how adding parens or hypens to the dropped text prevents that misreading.

My point with the suggestion (and my understanding of Malcolm's suggested wording) is this rule applies to Hollerith literal constants (in code) and Hollerith literal constants appearing inside an edit descriptor, where I _assume_ the latter still qualify as "literal constants"... but I don't have a copy of the ancient Fortran spec that supported these to verify that reading.

Reply:
Yeah, the assumption is not correct. They've been called out separately as "edit descriptors" since Fortran 77. (In Fortran 66 they were called "field descriptors".) The modern standard has them as such in subclause 13.3.2. Now, it _does_ say in R1322 that a _char-string-edit-desc_ is  _char-literal-constant_. But they are semantically distinct concepts still in the standard, even if they are syntactically identical. So I want to leave the distinction.

Reply:
We could possibly resolve the potential misparse and explcitly include edit descriptors by pedantically breaking it into two sentences:

```suggestion
literal constants that the processor supports.
FPP will not expand tokens in any alternative syntax for character
string edit descriptors that the processor supports.
```

Reply:
Relevant text from B.1-5:

> H edit descriptor.
> In Fortran 77 and Fortran 90, there was an alternative form of character string edit descriptor,
> which had been the only such form in Fortran 66; this has been deleted. A similar result can be
> achieved by using a character string edit descriptor.

Reply:
Yes. This is why Steve didn't want to mention it at all. It's been deleted for 30 years. (Taken to the extreme, need we mention that we will do macro expansion in arithmetic `IF`?  `PAUSE` statements? `ASSIGN` and assigned `GO TO` statements?)

Reply:
I agree it's a bit tangential, but the consensus from discussion was for us to specifically mention this one obsolescent feature.

Your thoughts on my latest proposal suggestion above?